README.md

```diff
@@ -194,6 +206,102 @@ default is 2GB, which is not enough to build p4c. Increase the memory limit to
 at least 4GB via Docker preferences or you are likely to see "internal compiler
 errors" from gcc which are caused by low memory.
 
+# Defining new CMake targets
+
+When building a new backend target, add it into the development tree in the
+extensions subdirectory. The cmake-based build system will automatically include
+it if it contains a CMakeLists.txt file.
+
+For a new backend, the cmake file should contain the following rules:
+
+## IR definition files
+
+Backend specific IR definition files should be added to the global list
+of IR_DEF_FILES as they are processed together with the core IR files.
+Use the following rule:
+
+```
+set (IR_DEF_FILES ${IR_DEF_FILES} ${MY_IR_DEF_FILES} PARENT_SCOPE)
+```
+where `MY_IR_DEF_FILES` is a list of file names with absolute path
+(for example, use `${CMAKE_CURRENT_SOURCE_DIR}`).
+
+If in addition you have additional supporting source files, they
+should be added to the frontend sources, as follows:
+
+```
+set(EXTENSION_FRONTEND_SOURCES ${EXTENSION_FRONTEND_SOURCES} ${MY_IR_SRCS} PARENT_SCOPE)
+```
+Again, `MY_IR_SRCS` is a list of file names with absolute path.
+
+## Source files
+
+Sources (.cpp and .h) should be added to the cpplint target using the following rule:
+
+```
+set (mybackend_CPPLINT_FILES ${MY_SOURCES_AND_HEADERS} PARENT_SCOPE)
+```
+
+where `mybackend` is the name of the directory you added under extensions.
+The p4c CMakeLists.txt will use that name to figure the full path of the files to lint.
+
+## Target
+
+Define a target for your executable. The target should link against
+the core `P4C_LIBRARIES` and `P4C_LIB_DEPS`.  `P4C_LIB_DEPS` are
```

Comment:
That's not true. For example, `libfrontend.la` depends on both `libcontrolplane.la` and `libp4ctoolkit.la`. `libcontrolplane.la` in turn depends on the Protobuf libraries. Handling that transparently is the reason we're using libtool in the current build system.